fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1850,39 +1909,47 @@ public static OnlineIndexer forRecordStoreAndIndex(@Nonnull FDBRecordStore recor
+         * and {@link #BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_ERROR_IF_POLICY_CHANGED}
+         *
          * <p>
          * Recommended. This should be sufficient if current index data is not corrupted.
          * </p>
          */
-        BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY(true),
+        BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY,
         /**
          * Build if the index is disabled; Rebuild if the index is write-only.
          */
-        BUILD_IF_DISABLED_REBUILD_IF_WRITE_ONLY(false),
+        BUILD_IF_DISABLED_REBUILD_IF_WRITE_ONLY,
         /**
          * Rebuild the index anyway, no matter it it disabled or write-only or readable.
          */
-        FORCE_BUILD(false),
+        FORCE_BUILD,
         /**
-         * Error if the index is disabled, or continue to build if the index is write only. To use this option to build
-         * an index, one should mark the index as write-only and clear existing index entries before building. This
-         * option is provided to make {@link #buildIndexAsync()} (or its variations) behave same as what it did before
-         * version 2.8.90.0, which is not recommended. {@link #BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY} should be adopted instead.
+         * Continue to build only if the index is write only and the requested method matches the previous one.
+         * (Error if the index is disabled, or if the existing type stamp do not match the requested indexing policy)
+         * To use this option to build an index, one should mark the index as write-only and clear the existing
+         * index entries before building. This option is provided to make {@link #buildIndexAsync()} (or its
+         * variations) behave same as what it did before version 2.8.90.0, which is not recommended.
+         * {@link #BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY} should be adopted instead.
          */
-        ERROR_IF_DISABLED_CONTINUE_IF_WRITE_ONLY(true),
-        ;
-
-        private boolean continueIfWriteOnly;
-
-        IndexStatePrecondition(boolean continueIfWriteOnly) {
-            this.continueIfWriteOnly = continueIfWriteOnly;
-        }
+        ERROR_IF_DISABLED_CONTINUE_IF_WRITE_ONLY,
 
-        public boolean isContinueIfWriteOnly() {
-            return continueIfWriteOnly;
-        }
+        /**
+         * Build if the index is disabled; Continue build if the index is write-only and the requested
+         * method is the same as the previous one. Else throw a RecordCoreException exception.
+         */
+        BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_ERROR_IF_POLICY_CHANGED,
+        /**
+         * Build if the index is disabled; Continue build if the index is write-only - only if the requested
+         * method matches the previous one, else restart the built.
+         */
+        BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED,
```

Comment:
Is the placement of these new options correct? I would think it would be:

```java
BUILD_IF_DISABLED,
BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_ERROR_IF_POLICY_CHANGED,
BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY,
BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED,
FORCE_BUILD,
```

To meet the requirement of "ordered in such a way that later build options cover more configs than earlier ones". (I guess `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY` and `BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY_REBUILD_IF_POLICY_CHANGED` are technically interchangeable, though having it continue seems less "disruptive" than having it rebuild, hence my proposed ordering.)

Reply:
Reordered.